File: backend/src/main/java/com/campus/Campus_Connect/features/event/dto/response/UserAccessResponse.java
package com.campus.Campus_Connect.features.event.dto.response;

import com.campus.Campus_Connect.features.event.entity.enums.EventMemberRole;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserAccessResponse {

    private Integer id;

    private String name;

    private Integer courseId;

    private Integer admissionYear;

    private EventMemberRole role;

    private Boolean hasAccess;
}

File: android-app/app/src/main/java/com/example/campusconnect/feature/events/data/remote/response/UserAccessResponse.kt
package com.example.campusconnect.feature.events.data.remote.response

import com.google.gson.annotations.SerializedName

data class UserAccessResponse(
    @SerializedName("id")
    val id: Int,

    @SerializedName("name")
    val name: String,

    @SerializedName("courseId")
    val courseId: Int?,

    @SerializedName("admissionYear")
    val admissionYear: Int?
)